### DISCUSSION TITLE
[Q] does the server support https or is it planned to add

### DISCUSSION THREAD

**Discussion by mawi42** [mathieucarbou/ESPAsyncWebServer#138] [General]
no https and not planned!
you can have a look at https://github.com/hoeken/PsychicHttp and its Discord at https://discord.gg/TAQrTR3f9C
(i am one of the project's main contributors also)

**Comment by mawi42** [comment]
thank you for the fast feeback and clarification, actually I saw this lib before but since it seems "just" for ESP32 and not ESP8266 I did not consider to use this.

**Reply by mathieucarbou** [reply]
Esp8266 is EOL in 4 years so anyway if you need https just drop them ;-)
In this fork I will completely drop support for 8266 in version 4 for sure.

**Reply by mawi42** [reply]
thank you again for the info, you are right I should also start to give up on my old ESP8266 projects.

Looking again into the alternative PsychicHttp I saw and remembered the second reason why I did not select it - no template support which I use a lot.

**Reply by mathieucarbou** [reply]
Oh good point. I will add that to our task list!

**Reply by mawi42** [reply]
That sounds a little like you recommend to move with mid term perspective to PsychicHttp

**Reply by mathieucarbou** [reply]
yes! both servers are great but serve quite different purposes for now. If you are only using esp32 and no libraries using asynctcp and/or espasyncws and you are bot bound to then, psychic is a good option. Regarding http calls, it is faster. A little more memory hungry though.

**Comment by mathieucarbou** [comment]
I am reopening: this is a useful discussion for other people too ;-)

**Reply by DRSDavidSoft** [reply]
Can the template processing be done in PsychichHTTP using a Middleware of some sort, instead of a built in server feature?

This would be super useful if we could override how middleware works; thus adapting it to our custom template processing needs (e,g, think of conditions, inclusions, etc)

**Reply by mathieucarbou** [reply]
There is no need to use any of both. 
In ESPAsyncWS, template processing is done using the response api because it was done like that but the 2 things are completely separated in the code. Template processing does not need to be in the response object.
We could have a TemplatProcessor class which is outputing in the response object the result of processing a char* template with some arguments. What the template processor needs is a way to output its bytes. Only that.

**Reply by DRSDavidSoft** [reply]
That makes much more sense, and is an excellent design 👌 

This reminds me of this bit in PsychicHTTP's README:

> No templating system (anyone actually use this?)

I guess this can be changed in the future soon 😄

**Reply by mathieucarbou** [reply]
I think it will: there is a need. Also, a way to efficiently combine several resources: some people are splitting header / body / footer for whatever good reasons they have but want to serve them as one big file.
